e2e/tcproute_test.go

```diff
@@ -441,3 +443,231 @@ func defaultNetworkURL() string {
 func defaultSubnetworkURL() string {
 	return cloud.NewSubnetworksResourceID(testFlags.project, region, "default").SelfLink(meta.VersionGA)
 }
+
+func createManyHealthchecks(graphBuilder *rgraph.Builder, hcNum int, name string) ([]*cloud.ResourceID, error) {
+	var hcs []*cloud.ResourceID
+	var e error
+	for i := 0; i < hcNum; i++ {
+		hc, err := buildHealthCheck(graphBuilder, "hc-"+name+strconv.Itoa(i), 15)
+		errors.Join(e, err)
+		hcs = append(hcs, hc)
+	}
+	return hcs, e
+}
+
+func createManyBackendServicesWithHC(graphBuilder *rgraph.Builder, bsNum int, name string, hcs []*cloud.ResourceID) ([]*cloud.ResourceID, error) {
+	hcNum := len(hcs)
+	if len(hcs) < hcNum {
+		return nil, fmt.Errorf("createManyBackendServicesWithHC: not enough healthchecks: want %d, got %d", bsNum, hcNum)
+	}
+	var bss []*cloud.ResourceID
+	var e error
+	for i := 0; i < bsNum; i++ {
+		bs, err := buildBackendServiceWithLBScheme(graphBuilder, name+"-"+strconv.Itoa(i)+"-bs", hcs[i], "INTERNAL_SELF_MANAGED")
+		errors.Join(e, err)
+		bss = append(bss, bs)
+	}
+	return bss, e
+}
+
+func createTcprRules(bs1ID *cloud.ResourceID, routeCIDR string, bs2ID *cloud.ResourceID, routeCIDR2 string) []*networkservices.TcpRouteRouteRule {
+	return []*networkservices.TcpRouteRouteRule{
+		{
+			Action: &networkservices.TcpRouteRouteAction{
+				Destinations: []*networkservices.TcpRouteRouteDestination{
+					{
+						ServiceName: resourceSelfLink(bs1ID),
+						Weight:      10,
+					},
+				},
+			},
+			Matches: []*networkservices.TcpRouteRouteMatch{
+				{
+					Address: routeCIDR,
+					Port:    "80",
+				},
+			},
+		},
+		{
+			Action: &networkservices.TcpRouteRouteAction{
+				Destinations: []*networkservices.TcpRouteRouteDestination{
+					{
+						ServiceName: resourceSelfLink(bs2ID),
+						Weight:      10,
+					},
+				},
+			},
+			Matches: []*networkservices.TcpRouteRouteMatch{
+				{
+					Address: routeCIDR2,
+					Port:    "80",
+				},
+			},
+		},
+	}
+}
+
+func createTCPRoutes(t *testing.T, graphBuilder *rgraph.Builder, numTCPR int, name_prefix string, meshURL string, bss []*cloud.ResourceID) ([]*cloud.ResourceID, error) {
+	var tcprs []*cloud.ResourceID
+	var e error
+	for i := 0; i < 2*numTCPR; i += 2 {
+		tcprRules1 := createTcprRules(bss[i], "10.240."+strconv.Itoa(i)+".83/32", bss[i+1], "10.240."+strconv.Itoa(i+1)+".83/32")
```

Comment:
this is so complex why not have createTcprRule and call it twice with different params?

tcpRule1 :=- createTcprRule(bss[i], "10.240."+strconv.Itoa(i)+".83/32", 
tcpRule2 L= createTcprRule(bss[i+1], "10.240."+strconv.Itoa(i+1)+".83/32")
you can pout it in array if you need